observability/exporting.go

```diff
@@ -0,0 +1,146 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package observability
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	gcplogging "cloud.google.com/go/logging"
+	grpclogrecordpb "google.golang.org/grpc/observability/internal/logging"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// loggingExporter is the interface of logging exporter for gRPC Observability.
+// In future, we might expose this to allow users provide custom exporters. But
+// now, it exists for testing purposes.
+type loggingExporter interface {
+	// EmitGrpcLogRecord writes a gRPC LogRecord to cache without blocking.
+	EmitGrpcLogRecord(*grpclogrecordpb.GrpcLogRecord)
+	// Close flushes all pending data and closes the exporter.
+	Close() error
+}
+
+// globalLoggingExporter is the global logging exporter, may be nil.
+var globalLoggingExporter loggingExporter
```

Comment:
Should this be a field inside observabilityBinaryLogger instead?  I think if we minimize our use of globals, that would be better.